src/tools/rbd/action/Wnbd.cc

```diff
@@ -180,6 +183,12 @@ int execute_unmap(const po::variables_map &vm,
     if (r < 0) {
       return r;
     }
+  } else {
+    device_name.clear();
+    int r = get_image_or_snap_spec(vm, &device_name);
```

Comment:
I don't think this is right.  The problem seems to be that `Wnbd.cc` has no way to distinguish between a positional argument that is a device name and a positional argument that is a (partial) spec.  Is there some kind of `/dev`-like prefix that can be relied on on Windows?  Check out how this is handled in `Nbd.cc`.

Reply:
> The wnbd driver uses the image spec to identify the disk ("instance name"), so rbd-wnbd is currently reporting that as device name, having a separate column for the disk number:

Is mapping the same image twice on the same node allowed (it's definitely allowed for krbd and nbd and I don't see it obviously disallowed in for wnbd)?  If yes, it looks like there is currently no way to differentiate between such mappings?

Reply:
The wnbd driver returns an error when attempting to map the same image twice, which is why we didn't add an explicit check in rbd-wnbd.

```
PS C:\Users\Administrator> rbd-wnbd map -f rbd_win_10g --non-persistent -f --wnbd-log-level=10
2022-10-05T04:59:03.344Pacific Daylight Time 1  0 ceph version 18.0.0-133-ga13d8885a9f (a13d8885a9f9edeb16bc734e543eab43869c010a) quincy (dev), process rbd-wnbd, pid 7620
2022-10-05T04:59:03.344Pacific Daylight Time 1  0 rbd-wnbd: Mapping RBD image: rbd/rbd_win_10g
2022-10-05T04:59:03.473Pacific Daylight Time 1  0 libwnbd.dll!WnbdOpenAdapterEx DEBUG Opening WNBD adapter device: \\?\root#scsiadapter#0000#{949dd17c-b06c-4c14-8e13-f1a3a4a6dbcb}
2022-10-05T04:59:03.473Pacific Daylight Time 1  0 libwnbd.dll!WnbdCreate DEBUG Mapping device. Name=rbd/rbd_win_10g, Serial=, Owner=ceph-rbd-wnbd, BC=20971520, BS=512, RO=0, Flush=1, PerRes=1, Unmap=1, UnmapAnchor=0, MaxUnmapDescCount=1, Nbd=0.
2022-10-05T04:59:03.473Pacific Daylight Time 1  0 libwnbd.dll!WnbdIoctlCreate ERROR Could not create WNBD disk. Error: 183. Error message: Cannot create a file when that file already exists.

2022-10-05T04:59:03.473Pacific Daylight Time 1  0 libwnbd.dll!WnbdClose DEBUG Closing device
```

Reply:
Is this identity check based on `cfg->devpath` (i.e. on a full image spec)?  If that's the case, I think can just stop pretending that device names are a thing for wnbd:
- get rid of `device_name` variable
- call `get_image_or_snap_spec` for `image_name` unconditionally